text/0001-cdk-update.md

```diff
@@ -0,0 +1,299 @@
+`cdk deploy`.
+
+## FAQ
+
+### What are we launching today?
+
+The `cdk deploy --accelerated` and `cdk deploy --accelerated --watch` commands,
+with support for rapid update of Lambda function code, images for ECS and
+Fargate task definitions, and AWS StepFunction workflows.
+
+### Why should I use this feature?
+
+If you are developing a CDK application and want to publish your code changes
+without waiting for CloudFormation to perform a full, safe stack update, you
+should use this feature.
+
+### Does it work when my Lambda is in Go or Java?
+
+To an extent, yes, but there are caveats. In order to update the lambda code you
+will need to run your build process to create the jar or executable that you are
+uploading as lambda code.
+
+### Does this feature work with `aws-lambda-go` for compiled Lambdas?
+
+Yes, if your lambda is defined by an `aws-lambda-go`, `aws-lambda-python`, or
+`aws-lambda-nodejs` construct, the accelerate and watch capabilities will work
+within those source trees to automatically deploy changes to your code as you
+make it.
+
+### What about "other resource type"?
+
+Based on customer response, we will consider expanding the set of AWS resources
+and asset types implemented in the CDK directly. The plugin mechanism for
+resource refreshing will allow other resource types to be supported by either
+community or vendor addition.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+The overhead of developing a Lambda application using the CDK is significant,
+since each code change currently requires a CloudFormation stack update to apply
+new code, _or_ manually introducing drift to the application by inspecting the
+stack and manipulating the resources directly.
+
+The CDK accelerate tool will allow the CDK to handle the resource changes rather
+than this manual process, introducing some implicit safety and reducing the
+manual labor of the update.
+
+### Why should we _not_ do this?
+
+This solution has a risk of introducing a deployment tool that users might use
+to shortcircuit the safe CloudFormation deployment process. If a user runs CDK
+update on their production stack, it can perform updates without adequate stack
+update safety checks. Releasing a public tool with instant update capability
+into the CDK may not be the right way to make this functionality public. To
+mitigate this, `accelerate` only runs on stacks that have explicitly opted in to
+the capability.
+
+### What changes are required to enable this change?
+
+1. An implementation of the `cdk deploy --accelerated` command in the CDK
+   toolkit that can examine the Application stacks for updates that can be
+   applied and apply them.
+2. The CDK toolkit must be able to query a CDK resource for the set of
+   filesystem resources it must monitor for the `--watch` operation, and run a
+   filesystem monitor for that.
+
+### Is this a breaking change?
+
+No, it is not.
+
+### What are the drawbacks of this solution?
```

Comment:
Isn't stack drift another drawback since mutations are happening outside of CloudFormation? Drift isn't a big concern for development stacks, but there's nothing stopping anyone from doing it in prod stack. 

The main concern would be situations where the regular `cdk deploy` using CloudFormation fails due to out of stack mutations. Perhaps that's mitigated by restricting accelerated deployments to certain constructs and preventing resource deletion?

Reply:
It's both a drawback _and_ the primary mechanism of the tool, so it's a bit hard to say one way or the other. Bug and feature, if you will :)